cpp/tests/test_sde_sirs.cpp

```diff
@@ -0,0 +1,181 @@
+    }
+    return model;
+}
+
+TEST(TestSdeSirs, get_flows)
+{
+    // make two get_flows computations with mocked rng
+    ScopedMockDistribution<
+        testing::StrictMock<MockDistribution<mio::DistributionAdapter<std::normal_distribution<double>>>>>
+        normal_dist_mock;
+
+    EXPECT_CALL(normal_dist_mock.get_mock(), invoke)
+        .Times(testing::Exactly(6))
+        .WillOnce(testing::Return(1.))
+        .WillOnce(testing::Return(0.))
+        .WillOnce(testing::Return(1.))
+        .WillOnce(testing::Return(0.))
+        .WillOnce(testing::Return(1.))
+        .WillOnce(testing::Return(0.));
+
+    const Eigen::Vector3d y = Eigen::Vector3d::Constant(1);
+    Eigen::Vector3d flows   = Eigen::Vector3d::Constant(1);
+
+    // results contain two parts : deterministic + stochastic
+
+    ssirs_testing_model().get_flows(y, y, 0, flows);
+    auto expected_result = Eigen::Vector3d{1 + 2, 1, 0.25 + 1};
+    EXPECT_EQ(flows, expected_result);
+
+    ssirs_testing_model().get_flows(y, y, 0, flows);
+    expected_result = Eigen::Vector3d{1, 1 + 2, 0.25};
+    EXPECT_EQ(flows, expected_result);
+}
+
+TEST(TestSdeSirs, Simulation)
+{
+    // make a single integration step via a simulation
+    // this should overwrite the model step size
+    ScopedMockDistribution<
+        testing::StrictMock<MockDistribution<mio::DistributionAdapter<std::normal_distribution<double>>>>>
+        normal_dist_mock;
+
+    EXPECT_CALL(normal_dist_mock.get_mock(), invoke)
+        .Times(testing::Exactly(3))
+        .WillOnce(testing::Return(.5))
+        .WillOnce(testing::Return(.5))
+        .WillOnce(testing::Return(.5));
+
+    auto integrator = std::make_shared<mio::EulerIntegratorCore>();
+    auto sim        = mio::ssirs::Simulation(ssirs_testing_model(), 0, 1);
+    sim.set_integrator(integrator);
+    sim.advance(1);
+
+    EXPECT_EQ(sim.get_model().step_size, 1.0); // set by simulation
+
+    EXPECT_EQ(sim.get_result().get_num_time_points(), 2); // stores initial value and single step
+
+    auto expected_result = Eigen::Vector3d{0.5, 1.0, 1.5};
+    EXPECT_EQ(sim.get_result().get_last_value(), expected_result);
+}
+
+TEST(TestSdeSirs, FlowSimulation)
+{
+    // make a single integration step via a flow simulation
+    // this should overwrite the model step size
+    ScopedMockDistribution<
+        testing::StrictMock<MockDistribution<mio::DistributionAdapter<std::normal_distribution<double>>>>>
+        normal_dist_mock;
+
+    EXPECT_CALL(normal_dist_mock.get_mock(), invoke)
+        .Times(testing::Exactly(3))
+        .WillOnce(testing::Return(.5))
+        .WillOnce(testing::Return(.5))
+        .WillOnce(testing::Return(.5));
+
+    auto integrator = std::make_shared<mio::EulerIntegratorCore>();
+    auto sim        = mio::ssirs::FlowSimulation(ssirs_testing_model(), 0, 1);
+    sim.set_integrator(integrator);
+    sim.advance(1);
+
+    EXPECT_EQ(sim.get_model().step_size, 1.0); // set by simulation
```

Comment:
Don't understand why this is 1.0. Isn't the default 0.1? In the `model` and `Simulation`?